README.md

```diff
@@ -2,47 +2,62 @@
-| `api-token`                       | [Account API token](https://docs.codacy.com/codacy-api/api-tokens/#account-api-tokens) | `${{ secrets.CODACY_API_TOKEN }}`     |
-| `project-token`                   | [Project API token](https://docs.codacy.com/codacy-api/api-tokens/#project-api-tokens) | `${{ secrets.CODACY_PROJECT_TOKEN }}` |
-| `coverage-reports`                | Optional comma-separated list of reports to send                                       | `''`                                  |
-| `language`                        | Optionally force associating a language with your coverage report                      | `''`                                  |
-| `force-coverage-parser`           | Optionally force using a specific coverage report parser                               | `''`                                  |
+    > ⚠️ **Never write API tokens to your configuration files** and keep your API tokens well protected, as they grant owner permissions to your projects on Codacy.
+
+2.  [Generate a supported code coverage report](https://docs.codacy.com/coverage-reporter/#generating-coverage) on each push to your repository.
+
+3.  Add the following to a file `.github/workflows/codacy-coverage-reporter.yaml` in your repository, where `PATH_TO_REPORT` is the path to the coverage report:
+
+    ```yaml
+    name: Codacy Coverage Reporter
+
+    on: ["push"]
+
+    jobs:
+      codacy-coverage-reporter:
+        runs-on: ubuntu-latest
+        name: codacy-coverage-reporter
+        steps:
+          - uses: actions/checkout@v2
+          - name: Run codacy-coverage-reporter
+            uses: codacy/codacy-coverage-reporter-action@v1
+            with:
+              project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+              # or
+              # api-token: ${{ secrets.CODACY_API_TOKEN }}
+              coverage-reports: <PATH_TO_REPORT>
+              # or a comma-separated list for multiple reports
+              # coverage-reports: <PATH_TO_REPORT>, <PATH_TO_REPORT>
+    ```
+
+4.  Optionally, to add coverage results to GitHub status checks, [configure a quality gate for coverage](https://docs.codacy.com/repositories-configure/integrations/github-integration/#status-checks) and [enable sending status checks](https://docs.codacy.com/repositories-configure/integrations/github-integration/#status-checks) on Codacy.
+
+## Extra configurations
+
+The Codacy GitHub Action is a wrapper for running the [Codacy Coverage Reporter CLI](https://github.com/codacy/codacy-coverage-reporter#codacy-coverage-reporter) and supports a subset of the parameters available for the command `report`. In addition to `coverage-reports`, `project-token`, and `api-token`, described in the example above, you can set:
+
+* `language`: optionally force associating a language with your coverage report(s).
+* `force-coverage-parser`: optionally force using a specific coverage report parser.
```

Comment:
Since these options aren't mandatory, it already becomes implicit that they're optional. If we don't mention that, the descriptions become simpler (and it also becomes easier to write the descriptions for the `project-token` and `api-token`, which are mutually exclusive :slightly_smiling_face:).

A small stylistic detail regarding lists is that [we can normally do without the periods](https://learn.microsoft.com/en-us/style-guide/scannable-content/lists#punctuation) at the end of each list item (it's not wrong having the periods, but I do find this style more minimalist).

```suggestion
* `language`: force associating a language with your coverage report(s)
* `force-coverage-parser`: force using a specific coverage report parser
```

I checked the corresponding section for the [Codacy Analysis CLI GitHub Action](https://github.com/codacy/codacy-analysis-cli-action#extra-configurations) and I like the syntax that you used for describing each option here better. When you're tweaking the other section, you should apply the same syntax there as well. :nerd_face: